File: Introductory Problems/Repetitions/Repetitions.cpp
#include <iostream>
#include <vector>

using namespace std;

void solve(string &sequence)
{
    int maxRepetition = 1;
    int currentRepetition = 1;

    for (size_t i = 1; i < sequence.length(); ++i)
    {
        if (sequence[i] == sequence[i - 1])
        {
            currentRepetition++;
        }
        else
        {
            maxRepetition = max(maxRepetition, currentRepetition);
            currentRepetition = 1;
        }
    }

    maxRepetition = max(maxRepetition, currentRepetition);
    cout << maxRepetition;
}

int main()
{
    string seq;
    cin >> seq;
    solve(seq);
}

File: Introductory Problems/Repetitions/Repetitions.py
def solve(seq):
    maxSeq = 1
    currentSeq = 1
    for i in range(1, len(seq)):
        if seq[i] == seq[i - 1]:
            currentSeq += 1
        else:
            maxSeq = max(maxSeq, currentSeq)
            currentSeq = 1
    maxSeq = max(maxSeq, currentSeq)
    print(maxSeq)


if __name__ == "__main__":
    seq = input()
    solve(seq)
